lightning/src/chain/chaininterface.rs

```diff
@@ -252,12 +253,21 @@ impl<'a, CL: Deref<Target = ChainListener + 'a> + 'a> BlockNotifier<'a, CL> {
 	}
 
 	/// Register the given listener to receive events.
-	// TODO: unregister
 	pub fn register_listener(&self, listener: CL) {
 		let mut vec = self.listeners.lock().unwrap();
 		vec.push(listener);
 	}
 
+    /// Unregister the given listener to no longer
+    /// receive events.
+    ///
+    /// If the same listener is registered multiple times, unregistering
```

Comment:
Can you indicate that the comparison is by the pointer returned by the Deref trait implementation.